### PR TITLE
Update interoperability module constants and schemas - Closes #7555

### LINKED ISSUE #7555: Update interoperability module constants and schemas
### Description

- Add/Remove/Update constants according to https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#notation-and-constants
- Update outboxRootSchema
 ```js
const outboxRootSchema = {
    "type": "object",
    "required": ["root"],
    "properties": {
        "root": {
            "dataType": "bytes",
            "length": HASH_LENGTH, // Add fixed length
            "fieldNumber": 1
        }
    }
```
- Update `chainAccountSchema` - Remove networkID/chainID and update length of properties
- Update `channelSchema` - Update `messageFeeTokenID` to be bytes
- Update `validatorsSchema` update length of properties
- Update `ownChainAccountSchema` - Remove ID and replace it with `chainID` and update length of properties
- Update `terminatedStateAccountSchema` - Update length of properties
- Update `terminatedOutboxAccountSchema` - Update length of properties
- Update `chainIDSchema` - Replace `ID` with `chainID` as bytes
- Remove network substore completely

### Acceptance Criteria

- Make sure all the properties, schemas and constants are updated as per the updated LIP

### Additional Information

- Related LIP update https://github.com/LiskHQ/lips/pull/148
- Keep https://github.com/LiskHQ/lisk-sdk/issues/7534 and https://github.com/LiskHQ/lisk-sdk/issues/7520 in mind and if resolved the close these issues as well
- Relevant task - https://github.com/LiskHQ/lisk-sdk/issues/7608

### PR DESCRIPTION
**Pull request by sitetester**
### What was the problem?

This PR resolves #7555

### CODE REVIEW

File: framework/src/modules/interoperability/constants.ts [hunk 1]
```diff
@@ -21,10 +21,7 @@ export const MAINCHAIN_ID = 1;
 export const MAINCHAIN_ID_BUFFER = utils.intToBuffer(MAINCHAIN_ID, 4);
 // TODO: To be updated after token module update
 export const TOKEN_ID_LSK = Buffer.from([0, 0, 0, 1, 0, 0, 0, 0]);
-export const TOKEN_ID_LSK_MAINCHAIN = {
-	chainID: utils.intToBuffer(1, 4),
-	localID: utils.intToBuffer(0, 4),
-};
+export const TOKEN_ID_LSK_MAINCHAIN = Buffer.from('0x0000000000000000');
```
Comment (ishantiw) [inline framework/src/modules/interoperability/constants.ts:24]:
```suggestion
export const TOKEN_ID_LSK_MAINCHAIN = Buffer.from('0000000000000000', 'hex');
```

File: framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts [hunk 2]
```diff
@@ -216,10 +216,7 @@ export class SidechainRegistrationCommand extends BaseInteroperabilityCommand {
 		await chainSubstore.set(context, chainID, sidechainAccount);
 
 		// Add an entry in the channel substore
-		const messageFeeTokenID = {
-			chainID: utils.intToBuffer(1, 4),
-			localID: utils.intToBuffer(0, 4),
-		};
+		const messageFeeTokenID = Buffer.from('0000000000000011', 'hex');
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts:219]: where did you get this value? shouldn't this be `0000000100000000`

File: framework/src/modules/interoperability/schemas.ts [hunk 3]
```diff
@@ -320,19 +320,10 @@ export const registrationCCMParamsSchema = {
...
-				},
-				localID: {
-					dataType: 'bytes',
-					fieldNumber: 2,
-				},
-			},
+			dataType: 'bytes',
+			minLength: TOKEN_ID_LENGTH,
+			maxLength: TOKEN_ID_LENGTH,
+			fieldNumber: 4,
```
Comment (ishantiw) [inline framework/src/modules/interoperability/schemas.ts:326]:
```suggestion
			fieldNumber: 3,
```

File: framework/src/modules/interoperability/utils.ts [hunk 4]
```diff
@@ -634,28 +636,6 @@ export const initGenesisStateUtil = async (
...
-			);
-		}
-		if (tokenID.chainID.equals(MAINCHAIN_ID_BUFFER) && tokenID.localID.readInt32BE(0) !== 0) {
-			throw new Error(
-				`Local id corresponding to the channel data store key ${channelData.storeKey.toString(
-					'hex',
-				)} is not valid.`,
-			);
-		}
-
```
Comment (ishantiw) [inline framework/src/modules/interoperability/utils.ts:658]: I think you need to replace this condition with updated LIP

> Either `channel.messageFeeTokenID==TOKEN_ID_LSK` (corresponding to the LSK token), or `Token.getChainID(channel.messageFeeTokenID)` must be equal to `channelData.storeKey` or `ownChainAccount.chainID` (the message fee token must be a native token of either chains).

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts (hunk 2 above)
Comment (shuse2) [reply framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts:219]: btw mainchain ID / mainchain token ID should be computed like
```
const networkID = this._config.ownChainID.slice(0, 1);
// 3 bytes for remaining chainID bytes
const mainchainID = Buffer.concat([networkID, Buffer.alloc(CHAIN_ID_LENGTH - 1, 0)]);
const mainchainTokenID = Buffer.concat([mainchainID, Buffer.alloc(LOCAL_ID_LENGTH, 0)];
```

Comment (ishantiw) [reply framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts:219]: we can have a util func for this

File: framework/src/modules/interoperability/types.ts [hunk 5]
```diff
@@ -185,52 +186,34 @@ export interface OwnChainAccountJSON {
...
 }
 
-export interface Inbox {
+type InboxOutbox = {
 	appendPath: Buffer[];
 	size: number;
 	root: Buffer;
-}
+};
+export type Inbox = InboxOutbox;
```
Comment (mitsuaki-u) [inline framework/src/modules/interoperability/types.ts:194]: Can you explain the reason for these typing choices?

**Review by mitsuaki-u — COMMENTED** [review]
Besides other's comments LGTM

**Review by mitsuaki-u — APPROVED** [review]
🚀

File: framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts (hunk 2 above)
Comment (ishantiw) [reply framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts]: https://github.com/LiskHQ/lisk-sdk/issues/7701

**Pull request merged**